DeferArgs 0.3, just a few hours after 0.2, is now released <a href="http://cheeseshop.python.org/pypi/DeferArgs/0.3">here</a>. The only update is support for attemp/catch-all blocks without any @catch(type) handlers. In response to some people in #twisted@irc.freenode.net, here are examples that show what can be done more clearly.<br /><br />To write any function that can accept deferred arguments nicely:<br /><pre><br />@deferargs<br />def printArgs(*args, **kwargs):<br />    print args<br />    print kwargs<br /></pre><br /><br />To add psuedo-exception handlers and a psuedo-finally block to a @deferargs decorated function:<br /><pre><br />@deferargs<br />def printArgs(*args, **kwargs):<br />    print args<br />    print kwargs<br />    print "This one is important! ", kwargs['important']<br />@catch(KeyError)<br />def onKeyError(e):<br />    print "There was no important kwarg!"<br />@catch()<br />def onAnyError(e):<br />    print "Crap!"<br />@cleanup()<br />def _(r):<br />    print "Every thing is done."<br /></pre><br /><br />There is also a model for a psuedo-try block, which basically decorates just calls the function immediately when you define it and its catch/cleanup handlers:<br /><pre><br />@attempt<br />def this():<br />    assert False<br />@cleanup<br />def _(r)<br />    print "And thats it."<br /></pre>